### PR TITLE
Add additional status methods

### PR DESCRIPTION
**Pull request by dlamy**
@ericdcobb @jfairley @leafknode

Including activeCount and queuedCount, useful for providing current watchfolder state.

### CODE REVIEW

File: src/main/java/com/levelsbeyond/jpoller/BasePollManager.java [hunk 1]
```diff
@@ -128,13 +130,22 @@ public final void validateAndStart() throws Exception {
 		doValidate();
 
 		// configure thread pool
-		executorService = new ExecutorCompletionService<Result>(Executors.newFixedThreadPool(maxConcurrent));
+		executor = (ThreadPoolExecutor)Executors.newFixedThreadPool(maxConcurrent);
```
Comment (jfairley) [inline src/main/java/com/levelsbeyond/jpoller/BasePollManager.java]: Stinks that we have to cast here... I see the need though. (`getActiveCount()`)

It seems silly but I wonder if it'd be worthwhile to have a simple test that would just...

``` java
assertThat(Executors.newFixedThreadPool(maxConcurrent)).isInstanceOf(ThreadPoolExecutor.class);
```

I'd hate to upgrade to Java 8 and discover in production.

Comment (dlamy) [inline src/main/java/com/levelsbeyond/jpoller/BasePollManager.java]: Well unless they really fuck us over it's safe, here's what that `Executors` method is doing.  If there's real concern I would probably just replace with that implementation line.

```
public static ExecutorService newFixedThreadPool(int nThreads) {
    return new ThreadPoolExecutor(nThreads, nThreads,
                                  0L, TimeUnit.MILLISECONDS,
                                  new LinkedBlockingQueue<Runnable>());
}
```

Comment (jfairley) [inline src/main/java/com/levelsbeyond/jpoller/BasePollManager.java]: Doing that would be better. I love using these simple statics (also `Lists`, `Sets`) and how they hide the gritty details, but if we have to cast, it feels like we're losing the benefit.

**Comment by dlamy** [comment]
updated.

**Comment by jfairley** [comment]
:+1: safety first

**Comment by jfairley** [comment]
Ooh. Null-checks, too. Very nice.

**Pull request merged**